Detections/ThreatIntelligenceIndicator/URLEntity_PaloAlto.yaml

```diff
@@ -30,12 +30,15 @@ query: |
     | where TimeGenerated >= ago(dt_lookBack)
     // Select on Palo Alto logs
     | where DeviceVendor == "Palo Alto Networks"
-    | where DeviceEventClassID == 'url'		 
+    | where DeviceEventClassID == 'url'
+    //Uncomment the line below to only alert on allowed connections
+    //| where DeviceAction != "block-url"
     //Select logs where URL data is populated
-    | extend Url = columnifexists('RequestURL', "None")
-    | where Url != "None" and isnotempty(Url)
+    | extend Url = columnifexists("RequestURL", "None")
+    | extend Url = iif(Url == "", extract("([^\"]+)", 1, tolower(AdditionalExtensions)), trim('"', Url))
+    | extend Url = iif(Url !startswith "http://" and ApplicationProtocol != "ssl", strcat('http://', Url), iif(Url !startswith "https://" and ApplicationProtocol == "ssl", strcat('https://', Url), Url))

```

Comment:
recommend we use =~ on the ApplicationProtocol == "ssl", just in case the name case changes